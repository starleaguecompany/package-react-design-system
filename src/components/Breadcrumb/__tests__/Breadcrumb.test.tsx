import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Menu } from '@starleaguecompany/react-icons';
import { Breadcrumb, Icon } from '../../../index';

describe('Breadcrumb', () => {
  test('should render correctly with no props', () => {
    render(<Breadcrumb />);

    expect(screen.getByTestId('Breadcrumb')).toMatchSnapshot();
    expect(screen.getByTestId('Breadcrumb')).toHaveClass('container');
  });

  test('should render correctly with attributes', () => {
    render(<Breadcrumb id="test-id" className="test-class" />);

    expect(screen.getByTestId('Breadcrumb')).toMatchSnapshot();
    expect(screen.getByTestId('Breadcrumb')).toHaveClass('test-class');
    expect(screen.getByTestId('Breadcrumb')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly breadcrumb items with text', () => {
    render(
      <Breadcrumb>
        <Breadcrumb.Item href="/href_1">Breadcrumb 1</Breadcrumb.Item>
        <Breadcrumb.Item href="/href_2">Breadcrumb 2</Breadcrumb.Item>
        <Breadcrumb.Item href="/href_3">Breadcrumb 3</Breadcrumb.Item>
      </Breadcrumb>
    );

    expect(screen.getByTestId('Breadcrumb')).toMatchSnapshot();
    expect(screen.getAllByRole('link').length).toEqual(3);
    expect(screen.getAllByRole('link')[0]).toBeVisible();
    expect(screen.getAllByRole('link')[1]).toBeVisible();
    expect(screen.getAllByRole('link')[2]).toBeVisible();
    expect(screen.getAllByRole('link')[0]).toHaveAttribute('href', '/href_1');
    expect(screen.getAllByRole('link')[1]).toHaveAttribute('href', '/href_2');
    expect(screen.getAllByRole('link')[2]).toHaveAttribute('href', '/href_3');
  });

  test('should render correctly breadcrumb items with icon', () => {
    render(
      <Breadcrumb>
        <Breadcrumb.Item href="/href_1">
          <Icon icon={<Menu />} />
          Breadcrumb 1
        </Breadcrumb.Item>
        <Breadcrumb.Item href="/href_2">
          <Icon icon={<Menu />} />
          Breadcrumb 2
        </Breadcrumb.Item>
      </Breadcrumb>
    );

    expect(screen.getByTestId('Breadcrumb')).toMatchSnapshot();
    expect(screen.getAllByRole('link').length).toEqual(2);
    expect(screen.getAllByTestId('Menu')[0]).toBeVisible();
    expect(screen.getAllByTestId('Menu')[1]).toBeVisible();
    expect(screen.getByText('Breadcrumb 1')).toBeVisible();
    expect(screen.getByText('Breadcrumb 2')).toBeVisible();
    expect(screen.getAllByRole('link')[0]).toHaveAttribute('href', '/href_1');
    expect(screen.getAllByRole('link')[1]).toHaveAttribute('href', '/href_2');
  });
});
