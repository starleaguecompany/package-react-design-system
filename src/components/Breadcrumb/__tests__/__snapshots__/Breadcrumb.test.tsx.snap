// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render correctly breadcrumb items with icon 1`] = `
<div
  class="container container direction-horizontal align-center"
  data-qa="Breadcrumb"
  style="column-gap: 8px; row-gap: 8px;"
>
  <a
    class="item container color-black"
    href="/href_1"
  >
    <span
      class="container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Menu"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5h12M4 10h12M4 15h12"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
    Breadcrumb 1
  </a>
  <span
    class="container size-16 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="ArrowRight"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m7 16 6-6-6-6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  <a
    class="item container color-black"
    href="/href_2"
  >
    <span
      class="container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Menu"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5h12M4 10h12M4 15h12"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
    Breadcrumb 2
  </a>
</div>
`;

exports[`Breadcrumb should render correctly breadcrumb items with text 1`] = `
<div
  class="container container direction-horizontal align-center"
  data-qa="Breadcrumb"
  style="column-gap: 8px; row-gap: 8px;"
>
  <a
    class="item container color-black"
    href="/href_1"
  >
    Breadcrumb 1
  </a>
  <span
    class="container size-16 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="ArrowRight"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m7 16 6-6-6-6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  <a
    class="item container color-black"
    href="/href_2"
  >
    Breadcrumb 2
  </a>
  <span
    class="container size-16 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="ArrowRight"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m7 16 6-6-6-6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  <a
    class="item container color-black"
    href="/href_3"
  >
    Breadcrumb 3
  </a>
</div>
`;

exports[`Breadcrumb should render correctly with attributes 1`] = `
<div
  class="test-class container container direction-horizontal align-center"
  data-qa="Breadcrumb"
  id="test-id"
  style="column-gap: 8px; row-gap: 8px;"
/>
`;

exports[`Breadcrumb should render correctly with no props 1`] = `
<div
  class="container container direction-horizontal align-center"
  data-qa="Breadcrumb"
  style="column-gap: 8px; row-gap: 8px;"
/>
`;
